### DISCUSSION TITLE
How do you create swf/fla files for contributing tests?

### DISCUSSION THREAD

**Discussion by Mai-Lapyst** [ruffle-rs/ruffle#5455] [Q&A] [answered]
I wanted to begin contributing ruffle by adding more tests to it, but after some quite extensive research (google) how to create swf's in 2021, im left with nothing much (since most of the software ist outdated by years or simply not available anymore (Adobe)). My question is how you all create custom swf's for testing with ruffle? What are open source tools in generel for flash development in 2021?

**Comment by danielhjacobs** [comment, marked as answer]
https://github.com/ruffle-rs/ruffle/blob/master/CONTRIBUTING.md#test-guidelines

**Reply by Mai-Lapyst** [reply]
Thanks, i've completly missed that part of the file xD
